src/microbe_stage/organelle_components/ChemoreceptorComponent.cs

```diff
@@ -107,6 +107,18 @@ public class ChemoreceptorUpgrades : IComponentSpecificUpgrades
     public float SearchAmount { get; set; }
     public Color LineColour { get; set; }
 
+    public bool Equals(IComponentSpecificUpgrades other)
+    {
+        if (other is not ChemoreceptorUpgrades otherChemoreceptor)
+            return false;
+
+        return TargetCompound?.InternalName == otherChemoreceptor.TargetCompound?.InternalName
```

Comment:
```suggestion
        return TargetCompound == otherChemoreceptor.TargetCompound
```

Compounds should be compared by reference as is done everywhere else in the codebase.